test_entities.py

```diff
@@ -81,15 +96,38 @@ def test_text() -> None:
 
 
 def test_symbol() -> None:
-    symbol = Symbol('01b03c10-7334-4bd5-b2bc-942c18325d2b', Name('Sym name'), Description('A multiline description.\n\nDescription'), Keywords('my, keywords'), Author('Test'), Version('0.2'), Created('2018-10-17T19:13:41Z'), Category('d0618c29-0436-42da-a388-fdadf7b23892'))
-    symbol.add_pin(SymbolPin('6da06b2b-7806-4e68-bd0c-e9f18eb2f9d8', Name('1'), Position(5.08, 20.32), Rotation(180.0), Length(3.81)))
+    symbol = Symbol(
+        '01b03c10-7334-4bd5-b2bc-942c18325d2b',
+        Name('Sym name'),
+        Description('A multiline description.\n\nDescription'),
+        Keywords('my, keywords'),
+        Author('Test'),
+        Version('0.2'),
+        Created('2018-10-17T19:13:41Z'),
+        Deprecated(False),
+        GeneratedBy('black magic'),
+        [Category('d0618c29-0436-42da-a388-fdadf7b23892')],
+    )
+    symbol.add_pin(SymbolPin(
+        '6da06b2b-7806-4e68-bd0c-e9f18eb2f9d8',
+        Name('1'),
+        Position(5.08, 20.32),
+        Rotation(180.0),
+        Length(3.81),
+        NamePosition(1.0, 2.0),
+        NameRotation(270.0),
+        NameHeight(2.5),
+        NameAlign('left center'),
+    ))
     polygon = Polygon('743dbf3d-98e8-46f0-9a32-00e00d0e811f', Layer('sym_outlines'), Width(0.25), Fill(False), GrabArea(True))
     polygon.add_vertex(Vertex(Position(-2.54, 22.86), Angle(0.0)))
     polygon.add_vertex(Vertex(Position(-2.54, -25.4), Angle(0.0)))
     polygon.add_vertex(Vertex(Position(-2.54, 22.86), Angle(0.0)))
     symbol.add_polygon(polygon)
     symbol.add_circle(Circle('b5599e68-ff6a-464b-9a40-c6ba8ef8daf5', Layer('sym_outlines'), Width(0.254), Fill(False), GrabArea(False), Diameter(1.27), Position(5.715, 0.0)))
     symbol.add_text(Text('b9c4aa19-0a46-400c-9c96-e8c3dfb8f83e', Layer('sym_names'), Value('{{NAME}}'), Align('center bottom'), Height(2.54), Position(0.0, 22.86), Rotation(0.0)))
+    symbol.add_approval('(approval foo)')
+    symbol.add_approval('(approval bar)')
```

Comment:
I don't like that we need to pass `(approval bar)` here. I'd prefer `Approval('bar')` with `Approval` as it's own entity.